invenio_app_rdm/records_ui/utils.py

```diff
@@ -55,3 +58,33 @@ def set_default_value(record_dict, value, path, default_prefix="metadata"):
         path = "{}.{}".format(default_prefix, path)
 
     dict_set(record_dict, path, value)
+
+
+def get_external_resources(record):
+    """Generate external resources from the record."""
+    result = []
+    resources_cfg = current_app.config["APP_RDM_RECORD_LANDING_PAGE_EXTERNAL_LINKS"]
+    for cfg in resources_cfg:
+        try:
+            res = cfg["render"](record)
+            if res:
+                result.append(res)
+        except Exception:
```

Comment:
The issue is that the passed `render` functions just receive a record, but are free to do any kind of manipulation they want (could e.g. make service calls to fetch additional info). If this was a clear-cut API in terms of accessed layers/APIs, we could narrow it down, but for now, it acts as a fail-safe so that we don't 500 on rendering the record landing page.